### ISSUE TITLE
Server Restart / boot players breaks save files

### ISSUE THREAD

**Issue by joldham23** [blaxxun-boop/ServerCharacters#80]
I've been using this mod for a couple of years now, so thank you for it. We've been running into a problem where, if a player isn't logged out by the time we do a server restart and gets booted, their character game save breaks and they're prompted to start a new character. We haven't been able to recover any characters that this has happened to. Once it happens, they just won't be able to use that character and either start over, or stop playing out of frustration. Any insight would be greatly appreciated! Thanks again!

**Comment by unlolicon** [comment]
Same issue

**Comment by blaxxun-boop** [comment]
No way to help without log files.

My best guess without any logs would be that you are using no map printer.

**Comment by unlolicon** [comment]
where do i find log files ?
and what mods qualify as a map printer ?

**Comment by blaxxun-boop** [comment]
BepInEx/LogOutput.log is the log file.
And the name of the mod is NoMapPrinter and is known for breaking save files.

**Comment by unlolicon** [comment]
[LogOutput.log](https://github.com/user-attachments/files/17090916/LogOutput.log)
![Mod list](https://github.com/user-attachments/assets/33f42290-f256-4cbd-8f10-7575709450ce)

**Comment by blaxxun-boop** [comment]
You are using crossplay with mods that do not support crossplay. Disable crossplay to fix your issue.

**Comment by unlolicon** [comment]
Same issue is occuring, i disabled earlier Almanac and Odin training place should i try reactivating them ?

**Comment by unlolicon** [comment]
[log.txt](https://github.com/user-attachments/files/17090947/log.txt)

**Comment by silenthill007** [comment]
[LogOutput.log](https://github.com/user-attachments/files/18307096/LogOutput.log)
https://pastebin.com/p46gq6gN
The first part of the text is an error that appears when I try to save the game after a long time, there is no specific trigger, I can save right after restarting the game - after a longer while I can't. The second part, which starts with --- Exception Handler --- appears when I click Exit Game.
[14:54]
I can see "servercharacters" so the second part is related with this mod, I'm not sure what's going on with first part.

**Comment by blaxxun-boop** [comment]
@silenthill007 Remove UnityDebuggerAssistant and send a new log, please.

**Comment by silenthill007** [comment]
[LogOutput.log](https://github.com/user-attachments/files/18308511/LogOutput.log)
It's fine now?

**Comment by blaxxun-boop** [comment]
@silenthill007 Yes, much more readable, thanks.
The issue comes very likely from a mod that adds custom skills.
[Info   :   BepInEx] Loading [SkillInjectorMod 1.1.1] seems suspicious in that case.

Please also note that Better Networking is well known to cause all kind of weird issues.

**Comment by silenthill007** [comment]
Yes, I deleted better networking. Also deleting ServerCharacters fixed an issue with closing the game.

**Comment by silenthill007** [comment]
So, removing SkillInjector and VitalityRewrite didn't help, I also turned off the skill sorting mod, turned on Professions and a few mods related to specific professions, more specifically: Cooking, Blacksmithing, Ranching, Farming, Mining, Lumberjacking, Building, Sailing (this mod also bugged at some point, despite setting the experience gain to a strong minimum, while traveling by boat I suddenly started receiving notifications that I was leveling up, but nothing changed in the skill table)
